Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/runtime/helper_test.go

```diff
@@ -63,6 +63,74 @@ func TestExtractList(t *testing.T) {
 	}
 }
 
+func TestExtractListGeneric(t *testing.T) {
+	pl := &api.List{
+		Items: []runtime.Object{
+			&api.Pod{ObjectMeta: api.ObjectMeta{Name: "1"}},
+			&api.Service{ObjectMeta: api.ObjectMeta{Name: "2"}},
+		},
+	}
+	list, err := runtime.ExtractList(pl)
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+	if e, a := len(list), len(pl.Items); e != a {
+		t.Fatalf("Expected %v, got %v", e, a)
+	}
+	if obj, ok := list[0].(*api.Pod); !ok {
+		t.Fatalf("Expected list[0] to be *api.Pod, it is %#v", obj)
+	}
+	if obj, ok := list[1].(*api.Service); !ok {
+		t.Fatalf("Expected list[1] to be *api.Service, it is %#v", obj)
+	}
+}
+
+type fakePtrInterfaceList struct {
+	Items []*runtime.Object
+}
+
+func (f fakePtrInterfaceList) IsAnAPIObject() {}
+
+func TestExtractListOfInterfacePtrs(t *testing.T) {
+	pl := &fakePtrInterfaceList{
+		Items: []*runtime.Object{},
+	}
+	list, err := runtime.ExtractList(pl)
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+	if len(list) > 0 {
+		t.Fatalf("Expected empty list, got %#v", list)
+	}
+}
+
+type fakePtrValueList struct {
+	Items []*api.Pod
+}
+
+func (f fakePtrValueList) IsAnAPIObject() {}
+
+func TestExtractListOfValuePtrs(t *testing.T) {
+	pl := &fakePtrValueList{
+		Items: []*api.Pod{
+			{ObjectMeta: api.ObjectMeta{Name: "1"}},
+			{ObjectMeta: api.ObjectMeta{Name: "2"}},
+		},
+	}
+	list, err := runtime.ExtractList(pl)
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+	if e, a := len(list), len(pl.Items); e != a {
+		t.Fatalf("Expected %v, got %v", e, a)
+	}
+	for i := range list {
+		if obj, ok := list[i].(*api.Pod); !ok {
+			t.Fatalf("Expected list[%d] to be *api.Pod, it is %#v", i, obj)
+		}
+	}
+}
+
```

Comment:
It may be useful to add a test case where Items is a pointer and/or interface. Otherwise I don't see how these exercise the new code